### PR TITLE
Workqueue

### LINKED ISSUE #38: virt-handler retries creating a domain in case of error - even if TPR got deleted
In case that a VM fails to create the `virt-handler` tries to recreate the domain - even if the TPR got deleted in the meanwhile.

Reproduce by undefine the default network.

### PR DESCRIPTION
**Pull request by rmohr**
Switch virt-handler to workqueue.
Fixes #38

### CODE REVIEW

**Comment by rmohr** [comment]
Also added an integration test to mache sure we follow the cluster changes.

**Comment by rmohr** [comment]
```
Running Suite: Tests Suite
==========================
Random Seed: 1484153863
Will run 4 of 4 specs

••••
Ran 4 of 4 Specs in 11.816 seconds
SUCCESS! -- 4 Passed | 0 Failed | 0 Pending | 0 Skipped PASS
ok  	kubevirt.io/kubevirt/tests	11.890s
```

File: pkg/kubecli/kubecli.go [hunk 1]
```diff
@@ -97,34 +99,36 @@ func NewInformer(
 		ListerWatcher:    lw,
 		ObjectType:       objType,
 		FullResyncPeriod: resyncPeriod,
-		RetryOnError:     false,
+		RetryOnError:     true,
 
 		Process: func(obj interface{}) error {
 			// from oldest to newest
+			fmt.Println(obj.(cache.Deltas))
```
Comment (stu-gott) [inline pkg/kubecli/kubecli.go]: Is this for debugging?

Comment (rmohr) [inline pkg/kubecli/kubecli.go]: Yes, will remove it. Thx

Comment (rmohr) [inline pkg/kubecli/kubecli.go]: Done

**Pull request merged**